scripts/traccc_throughput_mt_plotter.py

```diff
@@ -0,0 +1,405 @@
+#!/usr/bin/env python3
+#
+# TRACCC library, part of the ACTS project (R&D line)
+#
+# (c) 2023 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+#
+
+# Python import(s).
+import argparse
+import csv
+
+# ROOT import(s).
+import ROOT
+
+def runOnInputs(inputs, func):
+    '''Function running another function on every row in all input CSVs
+
+    Argument(s):
+       inputs  -- List of CSV file name, platform name pairs
+       func    -- The function-like-object to run on every CSV row
+    '''
+
+    # Loop over the files.
+    for input in inputs:
+        # Open this particular file.
+        with open(input[0], 'r') as csvFile:
+            csvReader = csv.DictReader(csvFile)
+            # Process each row of the file.
+            for row in csvReader:
+                # Execute the received function on this row.
+                func(row, input)
+                pass
+            pass
+        pass
+    return
+
+def getSampleNames(inputs):
+    '''Get the union of "sample names" from the inputs
+
+    Argument(s):
+       inputs  -- List of CSV file name, platform name pairs
+
+    Return:
+       A list of unique sample names
+    '''
+
+    # The result list.
+    result = []
+
+    def collectDirectory(row, input):
+        '''Collect the directory names from the CSV rows
+
+        Argument(s):
+           row    -- A CSV row coming from a CSV DictReader
+           input  -- The CSV file name, platform name pair for the current row
+        '''
+        # Use the "result" variable from the parent's scope.
+        nonlocal result
+        # If this directory was not seen yet, add it to the result.
+        dirname = row['directory']
+        if not dirname in result:
+            result.append(row['directory'])
+            pass
+        return
+    
+    # Collect the directory names.
+    runOnInputs(inputs, collectDirectory)
+
+    # Return the list.
+    return result
```

Comment:
This way of doing things is pretty wasteful. Doesn't really matter for this particular use-case, I suppose, but the amount of unnecessary file reading being done here is quite large.

Reply:
The script takes <1 second to process 5 CSV files. So I'm not going to put effort into making this more efficient.